ruler-gradle-plugin/src/main/kotlin/com/spotify/ruler/plugin/dynamicfeature/DynamicFeatureHandler.kt

```diff
@@ -0,0 +1,48 @@
+/*
+* Copyright 2021 Spotify AB
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.spotify.ruler.plugin.dynamicfeature
+
+import com.android.build.api.dsl.ApplicationExtension
+import org.gradle.api.Project
+import java.io.File
+
+/** Responsible for handling dynamic feature modules for an app project. */
+class DynamicFeatureHandler {
+
+    /** Parses and returns the set of dynamic feature modules contained in the given [project]. */
+    fun parse(project: Project): List<String> {
+        val extension = project.extensions.getByType(ApplicationExtension::class.java)
+        @Suppress("UnstableApiUsage")
+        return extension.dynamicFeatures.map { feature ->
+            project.rootProject.project(feature).name
+        }
+    }
+
+    /** Groups all given [apks] based on the dynamic feature that they belong to. */
+    fun groupApks(dynamicFeatures: List<String>, apks: List<File>): Map<String, List<File>> {
+        val keys = (dynamicFeatures + BASE_FEATURE_NAME).sortedByDescending(String::length) // Sort to avoid ambiguity
+        return apks.groupBy { apk ->
+            keys.firstOrNull { apk.name.startsWith(it) } ?: throw IllegalArgumentException("Could not group $apk")
```

Comment:
When we have dfm named "base-dfm", `startsWith` works not property.
You can refer to [this code](https://github.com/spotify/ruler/compare/main...BaeBae33:ruler:dfm#diff-2079660ca1e76b08e0294243229ac7c48ba0dff7e54637629bd78c8275cb4603R65-R73) for getting each module's apks. It refers to [ApkSetUtils](https://github.com/google/bundletool/blob/96669880398d3df3fc8cfe12c82f3e94fa7ef36e/src/test/java/com/android/tools/build/bundletool/testing/ApkSetUtils.java#L48). I think this way is stable.

Reply:
You're right, that's a good idea. Change it!